src/unstructured_client/_hooks/custom/split_pdf_hook.py

```diff
@@ -0,0 +1,505 @@
+            Tuple[str, requests.Session]: The initialized SDK options.
+        """
+        self.client = client
+        return base_url, client
+
+    def before_request(
+        self, hook_ctx: BeforeRequestContext, request: requests.PreparedRequest
+    ) -> Union[requests.PreparedRequest, Exception]:
+        """If `splitPdfPage` is set to `true` in the request, the PDF file is split into
+        separate pages. Each page is sent as a separate request in parallel. The last
+        page request is returned by this method. It will return the original request
+        when: `splitPdfPage` is set to `false`, the file is not a PDF, or the HTTP
+        has not been initialized.
+
+        Args:
+            hook_ctx (BeforeRequestContext): The hook context containing information about
+            the operation.
+            request (requests.PreparedRequest): The request object.
+
+        Returns:
+            Union[requests.PreparedRequest, Exception]: If `splitPdfPage` is set to `true`,
+            the last page request; otherwise, the original request.
+        """
+        operation_id = hook_ctx.operation_id
+        content_type = request.headers.get("Content-Type")
+        body = request.body
+        if not isinstance(body, bytes) or content_type is None:
+            return request
+
+        decoded_body = MultipartDecoder(body, content_type)
+        form_data = self._parse_form_data(decoded_body)
+        split_pdf_page = form_data.get(PARTITION_FORM_SPLIT_PDF_PAGE_KEY)
+        if split_pdf_page is None or split_pdf_page == "false":
+            return request
+
+        file = form_data.get(PARTITION_FORM_FILES_KEY)
+        if (
+            file is None
+            or not isinstance(file, File)
+            or not file.filename.endswith(".pdf")
+        ):
+            logger.warning("Given file is not a PDF. Continuing without splitting.")
+            return request
+
+        if self.client is None:
+            logger.warning("HTTP client not accessible! Continuing without splitting.")
+            return request
+
+        filename = file.filename.replace(".pdf", "")
+        pages = self._get_pdf_pages(file.content)
+        call_api_partial = functools.partial(
+            self._call_api,
+            request=request,
+            form_data=form_data,
+            filename=filename,
+        )
+        call_threads = self._get_split_pdf_call_threads()
+        self.partition_requests[operation_id] = []
+        with ThreadPoolExecutor(max_workers=call_threads) as executor:
+            for page_content, page_number, all_pages_number in pages:
+                self.partition_requests[operation_id].append(
+                    executor.submit(call_api_partial, (page_content, page_number))
+                )
+                if page_number == all_pages_number - 1:
+                    break
+
+        last_page_request = self._create_request(
+            request, form_data, next(pages)[:2], filename
```

Comment:
The `next(page)[:2]` looks a bit cryptic on first glance, maybe there's a nicer way to extract page_content and page_number, eg
```
last_page = next(pages)
last_page_content, last_page_number = last_page[:2]
``

Reply:
Changed as suggested